lib/validation.js

```diff
@@ -18,10 +18,12 @@
  *
  * @module numberValidation
  */
+const isDecimal = require('validator/lib/isDecimal');
+const isInt = require('validator/lib/isInt');
 
-module.exports = function numberValidation(input, settings) {
-  if (input.target.value === '' && !settings.target.empty) {
-    return `${input.target.name} cannot be left blank!`;
+module.exports = function numberValidation(input) {
+  if (input.target.value && !isInt(input.target.value.toString()) && !isDecimal(input.target.value.toString())) {
```

Comment:
mergy mcmergestein please!

On Tue, Sep 13, 2016 at 2:39 PM, Sam Richard <notifications@github.com>
wrote:

> In lib/validation.js
> <https://github.com/punchcard-cms/input-plugin-number/pull/52#discussion_r78618981>
> :
>
> >
> > -module.exports = function numberValidation(input, settings) {
> > -  if (input.target.value === '' && !settings.target.empty) {
> > -    return `${input.target.name} cannot be left blank!`;
> > +module.exports = function numberValidation(input) {
> > +  if (input.target.value && !isInt(input.target.value.toString()) && !isDecimal(input.target.value.toString())) {
>
> Really? Character count? If we're going to play that, why isn't value
> stored as a var and referenced?
>
> I understand what's up. My logic brain clearly isn't functioning today.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/punchcard-cms/input-plugin-number/pull/52/files/d7f0decc987f2d04a6b2a8cb6a2ac9a654e308c1#r78618981>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AANPY8PaT-XVYwDyG4PpbY23lEYK-LvGks5qpu3igaJpZM4J6ySo>
> .
>